### PR TITLE
:sparkles: feat(envtest): checkout k8s binaries to explicit location

### LINKED ISSUE #2389: EnvTest setup docs need to be updated
### What do you want to happen?

The ENV TEST was changed. So, we need to update the docs to clarifies that as add the new instructions.
See: https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest#where-does-it-put-all-those-binaries

Then, would be very nice instead of removing the current instructions make it clear when it loses its supports. 
 
Also, note that we probably have related steps in the migration guide from v2 to v3 which seems that needs to be revisited. 

### Extra Labels

/kind documentation

### PR DESCRIPTION
**Pull request by rumstead**
# Description
The controller runtime allows us to check out the binary assets to [an explicit location ](https://github.com/kubernetes-sigs/controller-runtime/blob/bcde6f084dd15ef93bee2a2e4668184e7472eddd/tools/setup-envtest/main.go#L50) [instead of specific locations on each OS](https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest#where-does-it-put-all-those-binaries). 

It's clear and obvious where the binaries are downloaded and would get automatically cleaned up if run in a CI pipeline. Additionally, other binaries are downloaded to the same `LOCALDIR`

Motivates by the comment: https://github.com/kubernetes-sigs/kubebuilder/issues/2389#issuecomment-1211133745

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @rumstead, 

Thank you for your contribution 🥇 
Note that this PR does not close the issue https://github.com/kubernetes-sigs/kubebuilder/issues/2389 because this one is about we update the docs. 

/ok-to-test

File: testdata/project-v2/api/v1/groupversion_info.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ limitations under the License.
 */
 
 // Package v1 contains API Schema definitions for the crew v1 API group
-//+kubebuilder:object:generate=true
-//+groupName=crew.testproject.org
+// +kubebuilder:object:generate=true
+// +groupName=crew.testproject.org
```
Comment (camilamacedo86) [inline testdata/project-v2/api/v1/groupversion_info.go:19]: @rumstead these changes do not seems related could you please revert?

**Review by camilamacedo86 — COMMENTED** [review]
@rumstead could you please squash the commits for we are able to move forward?

**Comment by rumstead** [comment]
> @rumstead could you please squash the commits for we are able to move forward?

Will do, but why not do that at merge time?

**Comment by camilamacedo86** [comment]
> Will do, but why not do that at merge time?
So that we have for example a clear and concise git history and if we need to revert the changes it can easily done as well.

File: testdata/project-v2/api/v1/groupversion_info.go (hunk 1 above)
Comment (camilamacedo86) [inline testdata/project-v2/api/v1/groupversion_info.go:19]: @rumstead could you pelase revert this change? 

Note that you need to run "make generate" for any PR that changes the scaffolds otherwise it will fail in our check done to ensure that sample testdata is updated accordingly : https://github.com/kubernetes-sigs/kubebuilder/runs/8071680207?check_suite_focus=true

Comment (rumstead) [reply testdata/project-v2/api/v1/groupversion_info.go:19]: that change is a result of make generate. 

> > Will do, but why not do that at merge time?
> > So that we have for example a clear and concise git history and if we need to revert the changes it can easily done as well.

Hmm, I don't follow how squashing on merge wouldn't satisfy that requirement. Maybe a convo for a different day. 
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

**Comment by rumstead** [comment]
> > Will do, but why not do that at merge time?
> > So that we have for example a clear and concise git history and if we need to revert the changes it can easily done as well.

Hmm, I don't follow how squashing on merge wouldn't satisfy that requirement. Maybe a convo for a different day. 
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

File: testdata/project-v2/api/v1/groupversion_info.go (hunk 1 above)
Comment (camilamacedo86) [reply testdata/project-v2/api/v1/groupversion_info.go:19]: Hi @rumstead,

> https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

It does not bring the same result.

Comment (rumstead) [reply testdata/project-v2/api/v1/groupversion_info.go:19]:
```
kubebuilder feat/add-envtest-path                                                                                                                                                                                                                                                                                                                                                                                                                                     
❯ git status
On branch feat/add-envtest-path
Your branch is up to date with 'origin/feat/add-envtest-path'.

nothing to commit, working tree clean

kubebuilder feat/add-envtest-path                                                                                                                                                                                                                                                                                                                                                                                                                                      
❯ make generate
rm -rf testdata/
./test/testdata/generate.sh
Building kubebuilder
...

kubebuilder feat/add-envtest-path                                                                                                                                                                                                                                                                                                                                                                                                                                     
❯ git status
On branch feat/add-envtest-path
Your branch is up to date with 'origin/feat/add-envtest-path'.

nothing to commit, working tree clean

```

**Comment by camilamacedo86** [comment]
HI @rumstead,

> Hmm, I don't follow how squashing on merge wouldn't satisfy that requirement. Maybe a convo for a different day.
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

That does not seem to ensure the same result. This project follows the same policies applied for the other projects under kubernetes-sig and Kubernetes repo which requires squashing the commits, see: https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits  


https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits

File: testdata/project-v2/api/v1/groupversion_info.go (hunk 1 above)
Comment (rumstead) [reply testdata/project-v2/api/v1/groupversion_info.go:19]: found it. thanks for the links!

Comment (rumstead) [reply testdata/project-v2/api/v1/groupversion_info.go]: seems to be a difference in go versions. I am on go1.19 but when I switch back to go1.18 the spaces are removed.

Comment (camilamacedo86) [reply testdata/project-v2/api/v1/groupversion_info.go]: @rumstead we do not support go 1.19 yet. 
That definitely shows to be the case. 
Thank you for checking that.

**Review by camilamacedo86 — APPROVED** [review]
The code changes for me shows fine and it is also passing in the ci test where we call make test, see: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kubebuilder/2896/pull-kubebuilder-test/1564277991389794304

So, I am ok to : 
/approved

However, I'd like to ask a help from others check this one as well.

@everettraven @FillZpp @erikgb

**Comment by erikgb** [comment]
Changes LGTM, but as commented by @camilamacedo86, this does not close https://github.com/kubernetes-sigs/kubebuilder/issues/2389 (missing doc update).

**Comment by rumstead** [comment]
> Changes LGTM, but as commented by @camilamacedo86, this does not close #2389 (missing doc update).

Happy to update the docs as well. Should I do it in this PR or another?

**Comment by camilamacedo86** [comment]
Hi @rumstead 

> Changes LGTM, but as commented by @camilamacedo86, this does not close https://github.com/kubernetes-sigs/kubebuilder/issues/2389 (missing doc update).
> Happy to update the docs as well. Should I do it in this PR or another?

The required changes in the doc seem out of the scope of this PR, would be great if we have a PR only for this one and get your help with it.
